src/components/dialog/demoOpenFromCloseTo/index.html

```diff
@@ -0,0 +1,26 @@
+<div ng-controller="AppCtrl" layout="row" ng-cloak style="height: 300px">
+  <div layout="column" layout-align="center center"
+       style="background-color: #f2f2f2" class="md-padding">
+    <span id="left">left</span>
+  </div>
+  <div layout="column" layout-align="center">
+    <p class="inset">
+      A dialog can specify its origin and target with <code>showFrom</code> and <code>closeTo</code> properties,
+      <code>showFrom</code> and <code>closeTo</code> both can get string(query selector), element or a rect.
```

Comment:
The options  <code>showFrom</code> and <code>closeTo</code> can be specified as a string [where internally `querySelector( )` is used to perform the DOM element lookup], element or an Rect object [with top, left, width, height fields].